File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.java
class OrderedStream {
    int ptr;
    String[] stream;
    public OrderedStream(int n) {
        this.stream = new String[n];
        this.ptr = 0;
    }
    
    public List<String> insert(int idKey, String value) {
        stream[idKey - 1] = value;
        List<String> list = new ArrayList<>();
        while(ptr < stream.length && stream[ptr] != null){
            list.add(stream[ptr]);
            ++ptr;
        }
        
        return list;
    }
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream obj = new OrderedStream(n);
 * List<String> param_1 = obj.insert(idKey,value);
 */

File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.py
class OrderedStream:

    def __init__(self, n: int):
        self.stream = [""] * (n+1)
        self.idx = 1

    def insert(self, idKey: int, value: str) -> List[str]:
        self.stream[idKey] = value
        curr = []
        while self.idx < len(self.stream) and self.stream[self.idx]:
            curr.append(self.stream[self.idx])
            self.idx += 1
        
        return curr


# Your OrderedStream object will be instantiated and called as such:
# obj = OrderedStream(n)
# param_1 = obj.insert(idKey,value)